docker/models/containers.py

```diff
@@ -170,10 +171,12 @@ def exec_run(self, cmd, stdout=True, stderr=True, stdin=False, tty=False,
         exec_output = self.client.api.exec_start(
             resp['Id'], detach=detach, tty=tty, stream=stream, socket=socket
         )
-        exit_code = 0
-        if stream is False:
-            exit_code = self.client.api.exec_inspect(resp['Id'])['ExitCode']
-        return (exit_code, exec_output)
+        if socket or stream:
+            return ExecResult(None, exec_output)
+        else:
```

Comment:
Hi @funkyfuture. It looks like the ``else`` block is unnecessary because of the ``return``. You can use something like:

```
if socket or stream:
    return ExecResult(None, exec_output)

return ExecResult(
    self.client.api.exec_inspect(resp['Id'])['ExitCode'],
    exec_output)
```

Regards.

Reply:
i prefer the explicit over the implicit.

Reply:
I think it looks cleaner, but it's a personal matter. Your code, your decision. Greetings.